src/util.ts

```diff
@@ -658,6 +658,42 @@ export async function handleFetch(request: string, options?: RequestInit) {
   return object;
 }
 
+/**
+ * Execute fetch and return object response, log if known error thrown, otherwise rethrow error.
+ *
+ * @param request - The request information.
+ * @param options - The fetch options.
+ * @param timeout - Timeout to fail request.
+ * @param errorCodesToCatch - array of error codes for errors we want to catch in a particular context
+ * @returns The fetch response JSON data or undefined (if error occurs).
+ */
+export async function fetchWithErrorHandling(
+  request: string,
+  options?: RequestInit,
+  timeout?: number | null,
+  errorCodesToCatch?: number[],
+) {
+  let result;
+  try {
+    if (timeout) {
+      result = Promise.race([
```

Comment:
Wouldn't this be the `response`? It seems like the logic is slightly different between `timeout` and no `timeout`, since `result = await response.json()` is not being called in this case.

Reply:
Good catch, both of these cases should use `handleFetch` which just returns `response.json()`.